### DISCUSSION TITLE
Handling MySQL datetime fields

### DISCUSSION THREAD

**Discussion by dlandtaa** [prisma/prisma#23070] [Q&A] [answered]
### Question

I have a MySQL database that contains a fair number of datetime columns. It turns out that MySQL datetime columns have no concept of a timezone, so it's basically up your code to assign it a timezone.

I am also using prisma for this project and it seems to want to assign those datetime fields the UTC timezone. This is not correct, the datetime field values are actually stored using another timezone.

Other mysql libraries like [this one](https://github.com/mysqljs/mysql) allow you specify a setting so that date fields come back from the database as strings rather than javascript Date objects. This lets me use libraries like moment or luxon to set the timezone to the correct one. However, I don't see a similar option for prisma.

How is this handled in prisma?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model MyTable {
    tableId       Int @id
    someField     Int 
    createdDate   DateTime
}
```

- OS: macOS
- Database: MySQL 5.7
- Node.js version: 20.11.0

```
prisma                  : 5.9.1
@prisma/client          : 5.9.1
Computed binaryTarget   : darwin-arm64
Operating System        : darwin
Architecture            : arm64
Node.js                 : v20.11.0
Query Engine (Node-API) : libquery-engine 23fdc5965b1e05fc54e5f26ed3de66776b93de64 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli 23fdc5965b1e05fc54e5f26ed3de66776b93de64 (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.9.0-32.23fdc5965b1e05fc54e5f26ed3de66776b93de64
Default Engines Hash    : 23fdc5965b1e05fc54e5f26ed3de66776b93de64
Studio                  : 0.497.0
```

**Comment by ludralph** [comment]
Hi @dlandtaa 👋 

Thank you for raising this question.

Prisma treats date values as UTC by default. This can cause issues when your MySQL database is configured with a non-UTC timezone. See this [issue](https://github.com/prisma/prisma/issues/5051). Currently, Prisma Client does not have a built-in feature to handle timezone for MySQL databases.

**Reply by dlandtaa** [reply, marked as answer]
Okay, thanks for the info. My approach for this is to use luxon to convert the timezone of the utc time that prisma is returning to the one I want without changing the time itself.

```
import { DateTime } from 'luxon';
myRow.myField = DateTime.fromJSDate(myRow.myField).toUTC().setZone('America/Chicago', { keepLocalTime: true });
```
